### PR TITLE
feat: ROI edge constraints for Neuprint Executor (#127)

### PR DESCRIPTION
**Pull request by jakobtroidl**
Neuprint edge constraints allow ROI specification now. Added functionality to motif_to_cyper in the Neuprint executor.

### CODE REVIEW

File: dotmotif/executors/NEUPRINT_JSON.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
+ATTRIBUTES = [
```
Comment (j6k4m8) [inline dotmotif/executors/NEUPRINT_JSON.py:1]: Is there a way to dynamically fetch this list in the `NeuPrintExecutor` constructor?

Comment (jakobtroidl) [reply dotmotif/executors/NEUPRINT_JSON.py]: yes, I am using the `[fetch_all_rois](https://github.com/connectome-neuprint/neuprint-python/blob/ba13264e4cc91954b31ea002b679f482c14dfb57/neuprint/queries/rois.py#L8)` function from neuprint-python now.

File: dotmotif/executors/NeuPrintExecutor.py [hunk 2]
```diff
@@ -118,5 +119,23 @@ def motif_to_cypher(
...
+                key = key.strip('\"')  # remove quotes if any
+                if key in NEUPRINT_JSON.ATTRIBUTES:
+                    for operator, values in constraints.items():
+                        for value in values:
+                            this_edge = """{}_{}["{}"] {} {}""".format(u, v, key, operator, str(value))
+                            that_edge_pre = """(apoc.convert.fromJsonMap({}.roiInfo)["{}"].pre {} {})""".format(u, key,
+                                                                                                                operator,
+                                                                                                                str(value))
+                            that_edge_post = """(apoc.convert.fromJsonMap({}.roiInfo)["{}"].post {} {})""".format(v,
+                                                                                                                  key,
```
[the remark this answers is not in the document]
Comment (jakobtroidl) [reply dotmotif/executors/NeuPrintExecutor.py]: done 👍

File: dotmotif/executors/NeuPrintExecutor.py [hunk 3]
```diff
@@ -118,5 +119,23 @@ def motif_to_cypher(
...
+        cypher = Neo4jExecutor.motif_to_cypher(motif, count_only, static_entity_labels)
+
+        for (u, v), a in motif.list_edge_constraints().items():
+            for key, constraints in a.items():
+                key = key.strip('\"')  # remove quotes if any
+                if key in NEUPRINT_JSON.ATTRIBUTES:
+                    for operator, values in constraints.items():
+                        for value in values:
+                            this_edge = """{}_{}["{}"] {} {}""".format(u, v, key, operator, str(value))
+                            that_edge_pre = """(apoc.convert.fromJsonMap({}.roiInfo)["{}"].pre {} {})""".format(u, key,
```
[the remark this answers is not in the document]
Comment (jakobtroidl) [reply dotmotif/executors/NeuPrintExecutor.py]: done 👍

File: dotmotif/executors/NeuPrintExecutor.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import pandas as pd
 from neuprint import Client
+from neuprint import fetch_all_rois
```
Comment (j6k4m8) [inline dotmotif/executors/NeuPrintExecutor.py:3]: ha, well that was a lot easier than having to implement it ourselves XD

**Comment by jakobtroidl** [comment]
> All tests are passing! This is an awesome addition, and it looks like it's ready to merge as far as I'm concerned. Is there anything else you want to add before we merge, @jakobtroidl? :)

please go ahead and merge it in :) Thanks, @j6k4m8.

**Pull request merged**